### PR TITLE
skip any recieved done_rows amounts and avoid creating extra resultsets

### LINKED ISSUE #66: Query result is not read from stored proc
My stored procedure looks like this:
```sql
CREATE PROCEDURE [dbo].[sp_NewItem] 
  @Param1 varchar(20),
  @Param2 datetime
AS
  insert into Items (Param1, Param2)
  values (@Param1, @Param2)

  return SCOPE_IDENTITY()
```
The code which calls that stored procedure is
```rust
        let result = client.query(
                "declare @rc int; exec @rc = sp_NewItem @P1, @P2; select @rc as Id", 
                &[&param1, &param2])
            .await?
            .into_row()
            .await?;
```
and the result set is empty.
The workaround I found is using 'nocount' outside a procedure call. Like this:
````
"set nocount on; declare @rc int; exec @rc = sp_NewItem @P1, @P2; select @rc as Id", 
````
So it seems that it reads number of affected rows from stream and stops. I also tried to update SP to use 'select' instead of 'return', but that doesn't change much. 
This proc is an old legacy database, but I've never had this problem before, working with other drivers. 
What do you think, is that a bug?

### PR DESCRIPTION
**Pull request by esheppa**
closes #66

### CODE REVIEW

**Comment by SpeedSX** [comment]
I would remove this part of statement in the test: "set nocount on". Test should pass without it if the fix is correct.

**Comment by SpeedSX** [comment]
> I would remove this part of statement in the test: "set nocount on". Test should pass without it if the fix is correct.

Nevermind, I jsut didn't notice there is a test with "nocount off" too...

**Comment by brokenthorn** [comment]
In which version and when will it be available, will this be included in? :D

**Comment by pimeys** [comment]
0.4.8 is out and should have this fix:

https://github.com/prisma/tiberius/blob/master/CHANGELOG.md

File: examples/async-std.rs [hunk 1]
```diff
@@ -1,14 +1,14 @@
+use async_std::net::TcpStream;
 use once_cell::sync::Lazy;
 use std::env;
 use tiberius::{Client, Config};
-use async_std::net::TcpStream;
 
 static CONN_STR: Lazy<String> = Lazy::new(|| {
-    env::var("TIBERIUS_TEST_CONNECTION_STRING")
-        .unwrap_or_else(|_| "server=tcp:localhost,1433;IntegratedSecurity=true;TrustServerCertificate=true".to_owned())
```
Comment (brokenthorn) [inline examples/async-std.rs:8]: why not unwrap_or()?

Comment (esheppa) [reply examples/async-std.rs:8]: This version keeps clippy happier :)

Comment (pimeys) [reply examples/async-std.rs:8]: `to_owned` is allocating memory and if using `unwrap_or`, it'll allocate even if we don't need it.

`unwrap_or_else` gets a lambda that allocates with the default value only if the env var is not set.

Comment (brokenthorn) [reply examples/async-std.rs:8]: > `to_owned` is allocating memory and if using `unwrap_or`, it'll allocate even if we don't need it.
> 
> `unwrap_or_else` gets a lambda that allocates with the default value only if the env var is not set.

I had no idea of this distinction! Thank you!

**Pull request merged**